src/datastore/postgres-store.ts

```diff
@@ -3756,6 +3761,71 @@ export class PgDataStore extends (EventEmitter as { new (): DataStoreEventEmitte
     });
   }
 
+  async updateBatchTokenOfferingLocked(client: ClientBase, lockedInfos: DbTokenOfferingLocked[]) {
+    const columnCount = 3;
+    const insertParams = this.generateParameterizedInsertString({
+      rowCount: lockedInfos.length,
+      columnCount,
+    });
+    const values: any[] = [];
+    for (const lockedInfo of lockedInfos) {
+      values.push(lockedInfo.address, lockedInfo.value, lockedInfo.block);
+    }
+    const insertQuery = `INSERT INTO token_offering_locked (
+      address, value, block
+      ) VALUES ${insertParams}`;
+    const insertQueryName = `insert-batch-subdomains_${columnCount}x${lockedInfos.length}`;
+    const insertLockedInfosQuery: QueryConfig = {
+      name: insertQueryName,
+      text: insertQuery,
+      values,
+    };
+    try {
+      const res = await client.query(insertLockedInfosQuery);
+      if (res.rowCount !== lockedInfos.length) {
+        throw new Error(`Expected ${lockedInfos.length} inserts, got ${res.rowCount}`);
+      }
+    } catch (e) {
+      logError(`Locked Info errors ${e.message}`, e);
+      throw e;
+    }
+  }
+
+  async getTokenOfferingLocked(address: string) {
+    return this.query(async client => {
+      const queryResult = await client.query<DbTokenOfferingLocked & { total: BigInt }>(
+        `
+         SELECT block, value, (SUM(value) OVER())::bigint AS total
+         FROM token_offering_locked
+         WHERE address = $1
+         GROUP BY id;
```

Comment:
What is the purpose of the `GROUP By id` statement?

Reply:
Initially I was using this query 
```
SELECT id, block, value, SUM(value)::bigint AS total
FROM token_offering_locked
WHERE address = $1
``` 
and it was giving me this error
`error: column "token_offering_locked.id" must appear in the GROUP BY clause or be used in an aggregate function`
So I added` GROUP BY`

But with this sum function `(SUM(value) OVER())::bigint AS total`, I think this `GROUP BY id` should be removed.